server/api.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+type API struct {
+	plugin *Plugin
+	router *mux.Router
+}
+
+func (a *API) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	a.router.ServeHTTP(w, r)
+}
+
+func (a *API) handlerRemoveAttachments(w http.ResponseWriter, r *http.Request) {
+	// Get post_id from the query parameters
+	err := r.ParseForm()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	postID := r.FormValue("post_id")
+	if !model.IsValidId(postID) {
+		http.Error(w, "Invalid post ID", http.StatusBadRequest)
+		return
+	}
+
+	// Check if the user is the post author or a system admin
+	userID := r.Header.Get("Mattermost-User-ID")
+	if !model.IsValidId(userID) {
+		http.Error(w, "Invalid user ID", http.StatusBadRequest)
+		return
+	}
+
+	post, appErr := a.plugin.API.GetPost(postID)
+	if appErr != nil {
+		http.Error(w, appErr.Error(), appErr.StatusCode)
+		return
+	}
+
+	if errReason := a.plugin.userHasRemovePermissionsToPost(userID, post.ChannelId, postID); errReason != "" {
+		http.Error(w, errReason, http.StatusForbidden)
+		return
+	}
+
+	// Otherwise, update the post without attachments and then soft-delete the attachments from the channel
+	originalFileIDs := post.FileIds
+
+	post.FileIds = []string{}
+	newPost, appErr := a.plugin.API.UpdatePost(post)
+	if appErr != nil {
+		http.Error(w, appErr.Error(), appErr.StatusCode)
+		return
+	}
+
+	// Soft-delete the attachments from channel
+	for _, fileID := range post.FileIds {
+		if err := a.plugin.SQLStore.DetachAttachmentFromChannel(fileID); err != nil {
+			a.plugin.API.LogError("error detaching attachment from channel", "err", err)
+			http.Error(w, "Internal server error, check logs", http.StatusInternalServerError)
+			return
+		}
+	}
+
+	// Attach the original file IDs to the original post so history is not lost
+	if err := a.plugin.SQLStore.AttachFileIDsToPost(newPost.OriginalId, originalFileIDs); err != nil {
```

Comment:
Files are still visible on the RHS of the channel and I think it's because we attach them to the original post ID that still belongs to the channel. 
0/5 if this is an issue or not. It depends on what's the product goal of the plugin.

![Screenshot 2024-08-22 at 12 29 20](https://github.com/user-attachments/assets/ed7e71eb-97b9-4407-a75a-876a248ee602)

Reply:
Ummmm, something should have changed, this one of the first things I tested myself. Let me check what's going on.

Reply:
Ok the problem is the query that removes the channelid from the file, is not updating the row nor giving an error. Investigating.

Reply:
https://github.com/mattermost/mattermost-plugin-attachments-remover/pull/1/commits/a6c4f06ddf2b26af63783e505a098a59187db94f